### PR TITLE
Guest to user cart

### PR DESCRIPTION
**Pull request by jhp0621**
Guest can now add items to the cart then once they log in/ sign up, they should see the guest items in their user cart! If they already had items the last time they logged in, those items should appear as well! If there are duplicate items, the larger quantity should show up :) 

-added a helper function called "guestToUser" that merges items in the guest cart to the user's cart once the guest logs in or signs up. (see server-> auth -> index.js (bottom of the file))

-changed price type from FLOAT to INTEGER and made sure AllPickles, SinglePickle, and Cart components display the correct price (divided it by 100 and fixed decimal place to 2).

-added Review section on the SinglePickle component (fetch single pickle api: now eager loads Review model)

### CODE REVIEW

File: client/components/SinglePickle.js [hunk 1]
```diff
@@ -28,11 +28,17 @@ class SinglePickle extends Component {
         <h1>{pickle.title} </h1>
         <img src={pickle.imageUrl} />
         <p> About this pickle: {pickle.description} </p>
-        <p> Price: ${pickle.price} </p>
+        <p> Price: ${(pickle.price / 100).toFixed(2)} </p>
         <button type="button" onClick={() => this.handleClick(pickle)}>
           {' '}
           Add to cart
         </button>
+        <p>
```
Comment (jennmez) [inline client/components/SinglePickle.js]: ideas for future updates on this component display: 
- display the nickname of the user/guest leaving the comment
- add fakerjs seed data that lists the content, adds associations between userId/guest(null) & pickleId
- display a "no review" div if there are no reviews associated to the pickle
- date/time or just date of review

File: client/components/Cart.js [hunk 2]
```diff
@@ -26,27 +26,27 @@ class Cart extends React.Component {
         {cart.map(item => (
           <div key={item.pickle.id}>
             <Link to={`/pickles/${item.pickle.id}`}> {item.pickle.title} </Link>
-            <p>Quantity: {item.quantity}</p>
+            <p>Quantity: {item.qty}</p>
```
Comment (jennmez) [inline client/components/Cart.js]: Maybe we can move QTY inbetween the "remove one"  & "add one" section so the + - is separated by the quantity

Comment (jhp0621) [reply client/components/Cart.js]: Yes I like that idea! I'll fix that :)

File: client/components/SinglePickle.js (hunk 1 above)
Comment (jhp0621) [reply client/components/SinglePickle.js]: Yup. All sounds good. It also probably makes sense to create a Review component and render it via SinglePickle. (I added the review section just to get things started). We can work on it after the admin stuff (from tier 1) is done :)

**Pull request merged**